scipy/interpolate/_bsplines.py

```diff
@@ -314,6 +322,14 @@ def __call__(self, x, nu=0, extrapolate=None):
         x = np.asarray(x)
         x_shape, x_ndim = x.shape, x.ndim
         x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+
+        # With periodic extrapolation we map x to the segment
+        # (self.t[k], self.t[-k]).
+        if extrapolate == 'periodic':
```

Comment:
Don't understand why you say mapping is done to `(self.t[k], self.t[-k])` but then you use `self.t[self.k + 1],  self.t[-self.k - 1]`.